### PR TITLE
update test_ordered_dict.py from cpython 3.11.2

### PR DESCRIPTION
**Pull request by Brain-organizer**
update test_ordered_dict.py from cpython 3.11.2

### CODE REVIEW

File: Lib/test/test_ordered_dict.py [hunk 1]
```diff
@@ -239,6 +239,8 @@ def test_popitem_last(self):
         obj.popitem(last=True)
         self.assertEqual(len(obj), 20)
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_ordered_dict.py:243]: Please break this out into a separate commit.

File: Lib/test/test_ordered_dict.py [hunk 2]
```diff
@@ -281,12 +283,16 @@ def test_equality(self):
         # different length implied inequality
         self.assertNotEqual(od1, OrderedDict(pairs[:-1]))
 
+    # TODO: RUSTPYTHON
+    @unittest.expectedFailure
```
Comment (fanninpm) [inline Lib/test/test_ordered_dict.py:287]: Please break this out into a separate commit.

**Review by fanninpm — CHANGES_REQUESTED** [review]

**Comment by DimitrisJim** [comment]
Thanks! The results have some failures and some successes. Failures:

```
test.test_ordered_dict.CSimpleLRUCacheTests.test_pop
test.test_ordered_dict.PySimpleLRUCacheTests.test_pop
```

and the unexpected successes:

```
test.test_ordered_dict.CPythonOrderedDictSubclassTests.test_copying
test.test_ordered_dict.CPythonOrderedDictSubclassTests.test_pop
test.test_ordered_dict.CPythonOrderedDictTests.test_copying
test.test_ordered_dict.CPythonOrderedDictTests.test_pop
test.test_ordered_dict.PurePythonOrderedDictSubclassTests.test_copying
test.test_ordered_dict.PurePythonOrderedDictSubclassTests.test_pop
test.test_ordered_dict.PurePythonOrderedDictTests.test_copying
test.test_ordered_dict.PurePythonOrderedDictTests.test_pop
```

**Comment by DimitrisJim** [comment]
Superseded by https://github.com/RustPython/RustPython/pull/4846, thanks for opening the PR though!

**Pull request closed without merging**